hudi-spark-datasource/hudi-spark2/src/main/scala/org/apache/spark/sql/adapter/Spark2Adapter.scala

```diff
@@ -186,4 +186,13 @@ class Spark2Adapter extends SparkAdapter {
     case OFF_HEAP => "OFF_HEAP"
     case _ => throw new IllegalArgumentException(s"Invalid StorageLevel: $level")
   }
+
+  override def translateFilter(predicate: Expression,
+                               supportNestedPredicatePushdown: Boolean = false): Option[Filter] = {
+    if (supportNestedPredicatePushdown) {
```

Comment:
Is this expected to fail any spark 2 queries ?

Reply:
Sounds good. Can you add them in the javadoc.

Reply:
I don't see this in javadoc here or in SparkAdapter. In general, please add as much javadoc/scaladoc as possible for public APIs for future reference.

Reply:
Oh, sorry, must be rebase making this missing, added it back.